internal/hook/wait_exec_hook_handler_test.go

```diff
@@ -0,0 +1,926 @@
+							Name: "container1",
+							State: v1.ContainerState{
+								Running: &v1.ContainerStateRunning{},
+							},
+						}).
+						ContainerStatuses(&v1.ContainerStatus{
+							Name: "container2",
+							State: v1.ContainerState{
+								Running: &v1.ContainerStateRunning{},
+							},
+						}).
+						Result(),
+				},
+			},
+			changes: []change{
+				// 1st modification: container1 starts running, resourceVersion 2, container2 still waiting
+				{
+					wait: 10 * time.Millisecond,
+					updated: builder.ForPod("default", "my-pod").
+						ObjectMeta(builder.WithResourceVersion("2")).
+						Containers(&v1.Container{
+							Name: "container1",
+						}).
+						Containers(&v1.Container{
+							Name: "container2",
+						}).
+						ContainerStatuses(&v1.ContainerStatus{
+							Name: "container1",
+							State: v1.ContainerState{
+								Running: &v1.ContainerStateRunning{},
+							},
+						}).
+						ContainerStatuses(&v1.ContainerStatus{
+							Name: "container2",
+							State: v1.ContainerState{
+								Waiting: &v1.ContainerStateWaiting{},
+							},
+						}).
+						Result(),
+				},
+				// 2nd modification: container2 starts running, resourceVersion 3
+				{
+					wait: 10 * time.Millisecond,
+					updated: builder.ForPod("default", "my-pod").
+						ObjectMeta(builder.WithResourceVersion("3")).
+						Containers(&v1.Container{
+							Name: "container1",
+						}).
+						Containers(&v1.Container{
+							Name: "container2",
+						}).
+						ContainerStatuses(&v1.ContainerStatus{
+							Name: "container1",
+							State: v1.ContainerState{
+								Running: &v1.ContainerStateRunning{},
+							},
+						}).
+						ContainerStatuses(&v1.ContainerStatus{
+							Name: "container2",
+							State: v1.ContainerState{
+								Running: &v1.ContainerStateRunning{},
+							},
+						}).
+						Result(),
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+
+			source := fcache.NewFakeControllerSource()
+			go func() {
+				source.Add(test.initialPod)
+				for _, change := range test.changes {
+					time.Sleep(change.wait)
+					source.Modify(change.updated)
+				}
+			}()
+
+			podCommandExecutor := &velerotest.MockPodCommandExecutor{}
+			defer podCommandExecutor.AssertExpectations(t)
+
+			h := &DefaultWaitExecHookHandler{
+				PodCommandExecutor: podCommandExecutor,
+				ListWatchFactory:   &fakeListWatchFactory{source},
+			}
+
+			for _, e := range test.expectedExecutions {
+				obj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(e.pod)
+				assert.Nil(t, err)
+				podCommandExecutor.On("ExecutePodCommand", mock.Anything, obj, e.pod.Namespace, e.pod.Name, e.name, e.hook).Return(e.error)
+			}
+
+			ctx := context.Background()
+			if test.sharedHooksContextTimeout > 0 {
+				ctx, _ = context.WithTimeout(ctx, test.sharedHooksContextTimeout)
+			}
+
+			// TODO get the byContainer map or specify it in the unit tests
```

Comment:
Is the TODO here meant to be done before merging this PR?

Reply:
That was leftover from when the tests were WIP and I was still passing nil for the map. Removed.